chassis/core/src/main/java/com/griddynamics/jagger/invoker/QueryPoolScenario.java

```diff
@@ -45,15 +45,19 @@ public QueryPoolScenario(Invoker<Q, R, E> invoker, Iterator<Pair<Q, E>> chunks,
         this.chunks = chunks;
         this.systemClock = systemClock;
     }
-
+    
     @Override
-    public void doTransaction() {
+    public boolean doTransaction() {
         Pair<Q, E> chunk = chunks.next();
+        if (chunk == null) {
```

Comment:
we are not supposed to know the exact reason  here. it just can by contract but exact reason is implementation specific